Lessson16/havva/script.js

```diff
@@ -0,0 +1,687 @@
+     - 12, 1, 2  -> "Winter"
+     - 3, 4, 5   -> "Spring"
+     - 6, 7, 8   -> "Summer"
+     - 9, 10, 11 -> "Autumn"
+   - Log the season or "Invalid month" if out of range.
+*/
+
+function getSeason(monthNum) {
+  switch (monthNum) {
+    case 12:
+    case 1:
+    case 2:
+      console.log('Winter');
+      break;
+    case 3:
+    case 4:
+    case 5:
+      console.log('Spring');
+      break;
+    case 6:
+    case 7:
+    case 8:
+      console.log('Summer');
+      break;
+    case 9:
+    case 10:
+    case 11:
+      console.log('Autumn');
+      break;
+
+    default:
+      console.log('Invalid month');
+      break;
+  }
+}
+
+getSeason(1);
+getSeason(4);
+getSeason(7);
+getSeason(10);
+getSeason(13);
+
+/*
+25. Check If String Contains Number
+   - Define a function `containsNumber(str)` that uses a loop or a method like
+     .match() to check if there's any digit in the string.
+   - Log "Contains number" or "No number found".
+*/
+
+function containsNumber(str) {
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i];
+
+    if (!isNaN(parseInt(char))) {
+      return console.log('Contains number');
+
+    }
+  }
+  console.log('No number found');
+}
+
+containsNumber('Hello123');
+containsNumber('NoDigitsHere');
+containsNumber('2024 is the year');
+containsNumber('Just some text');
+
+/*
+26. Pad a String
+   - Define a function `padString(str, maxLength)` that if str.length < maxLength,
+     uses .padEnd() or .padStart() to make the string reach maxLength with '*'.
+   - Log the padded string. 
+*/
+
+function padString(str, maxLength) {
+  if (str.length < maxLength) {
+    const paddedStr = str.padEnd(maxLength, '*');
+    console.log(paddedStr);
+  }
+  else {
+    console.log(str);
+  }
+}
+
+padString('Hello', 10);
+padString('This is a long string', 10);
+padString('Short', 8);
+
+
+/*
+27. If-Else: Voting Eligibility
+   - Define a function `canVote(age)` that logs:
+     - "Can vote" if age >= 18
+     - "Too young to vote" otherwise
+*/
+function canVote(age) {
+  if (age >= 18) {
+    console.log('Can vote');
+  }
+  else {
+    console.log('Too young to vote');
+  }
+}
+
+canVote(19);
+canVote(15);
+/*
+28. Reverse Words in a Sentence
+   - Define a function `reverseWords(sentence)` that:
+     - Splits the sentence by spaces
+     - Reverses each word individually
+     - Joins them back
+   - Log the result.
+*/
+function reverseWords(sentence) {
+  const words = sentence.split(' ');
+  for (let i = 0; i < words.length; i++) {
+    words[i] = words[i].split('').reverse().join('');
+  }
+  const reversedSentence = words.join(' ');
+  console.log(reversedSentence);
+}
+
+reverseWords('Hello world from JavaScript');
+reverseWords('I love coding');
+
+/*
+29. Check Substring Position
+   - Define a function `findWordPosition(sentence, word)` that uses .indexOf(word)
+     to find the starting index. If not found, return -1.
+   - Log the index or log "Not found" if it's -1. -- ask this 
```

Comment:
The function .indexOf() returns -1 if the searched value is not found. That's how it works. In JS, you can often see checks like `if (sentence.indexOf(word))` meaning if the is not found in the sentence